lib/base/convert.cpp

```diff
@@ -38,10 +38,20 @@ String Convert::ToString(double val)
 	double integral;
 	double fractional = std::modf(val, &integral);
 
-	if (fractional == 0)
-		return Convert::ToString(static_cast<long long>(val));
-
 	std::ostringstream msgbuf;
+
+	if (fractional == 0) {
+		if (val < UINT64_MAX) {
+			if (val >= 0) {
+				return Convert::ToString(static_cast<unsigned long long>(val));
+			} else {
+				return Convert::ToString(static_cast<long long>(val));
+			}
```

Comment:
Please update the code style to use shortened if statements, e.g

```c++
if (val >= 0)
    return Convert::ToString(static_cast<unsigned long long>(val));
else
    return Convert::ToString(static_cast<long long>(val));
```

Reply:
I consider this bad practice (my trainers at @NETWAYS taught me that).

Reply:
@Al2Klimov What exactly? We use shortened if statements almost everywhere, where it is possible. I am a bit surprised that this is now bad practice.

Reply:
I just said *I consider* that bad practice, not that from now *you'll have to* do it the other way. However I'm surprised about I seem to be the first wo said this.